crypto/bls.go

```diff
@@ -339,6 +348,15 @@ func (pk *PubKeyBLSBLS12381) Size() int {
 // Encode returns a byte encoding of the public key.
 // The encoding is a compressed encoding of the point
 // [zcash] https://github.com/zkcrypto/pairing/blob/master/src/bls12_381/README.md#serialization
+func (a *PubKeyBLSBLS12381) EncodeCompressed() []byte {
+	if serializationG2 != compressed {
+		return nil
```

Comment:
I think it's safer to panic in this case than to return nil. We could add another return value for errors but that would impact all the interface. 
I can add an issue to support both formats dynamically in the future, and then I'll remove the panic.   
```suggestion
		panic("library is not configured to use compressed public key serialization")
```

Reply:
🤔  I'm not a fan of introducing more panics than absolutely necessary _at all_, but since this is on a dead code path, and would probably be detected on unit tests alone ...

Reply:
I agree. I will think of a better way to avoid this, for now the panic is a way to flag a bad set up of the lib.